File: python/algorithm.py
# Binary Search (Logarithmic Time: O(log n))
# Best for searching element or index number in already sorted list and, input only known elements
someList = [1, 7, 16, 24, 37, 40, 68, 71, 82, 98]

def binarySearch(listQuery: list, query: int):
    firstBar = 0
    lastBar = len(listQuery)-1
    found = False

    while firstBar <= lastBar and found is False:
        midPointer = (firstBar + lastBar) // 2
        if query == listQuery[midPointer]: print("Found at index [",midPointer,"]",sep=''); found = True
        else:
            if query > listQuery[midPointer]: firstBar = midPointer + 1
            else: lastBar = midPointer - 1
        
binarySearch(someList, 82)

File: javascript/algorithm.js
// Placeholder Binary Search Algo.
let myList = [2, 12, 27, 34, 49, 51, 60, 76, 83, 88, 95, 101]

binarySearch(myList, 95)

function binarySearch(list, query) {
    firstBar = 0
    lastBar = list.length -1
    found = false
    
    while (firstBar <= lastBar && !found){
        middlePoint = Math.floor((firstBar + lastBar) / 2); middlePoint
        // found = true; break
        if (query == list[middlePoint]){
            console.log(`found at index [${ middlePoint }]`)
            found = true
        }
        else {
            if (query < list[middlePoint]) lastBar = middlePoint - 1
            else firstBar = middlePoint + 1
        }
    }
}
